src/Associations.cc

```diff
@@ -33,6 +34,8 @@ static double sqr(double x) {return x*x;}
 namespace lsst {
 namespace jointcal {
 
+static LOG_LOGGER _log = LOG_GET("jointcal.Associations");
```

Comment:
Why not instantiate in an anonymous namespace, like how qserv does? 

Same questions in other files.